service/hub/auth/authenticate.go

```diff
@@ -0,0 +1,53 @@
+package auth
+
+import (
+	"context"
+
+	"github.com/cesanta/glog"
+	"github.com/labring/sealos/pkg/client-go/kubernetes"
+	"github.com/labring/service/hub/api"
+)
+
+func init() {
+	glog.Info("authn plugin init function called")
+}
+
+type SealosAuthenticate struct {
+	api.Authenticator
+}
+
+func (a SealosAuthenticate) Authenticate(user string, password api.PasswordString) (bool, api.Labels, error) {
+	glog.Info("Authenticate for user:", user)
+	// todo replace server ip to env $(SERVER)
+
+	// create client
+	client, err := kubernetes.NewKubernetesClientByConfigString(string(password))
+	if err != nil {
+		glog.Error("NewKubernetesClientByConfigString error")
+		return false, api.Labels{}, api.ErrWrongPass
+	}
+	// check client by ping apiserver
+	// or get organizations?
+	res, err := client.Discovery().RESTClient().Get().AbsPath("/healthz").DoRaw(context.Background())
```

Comment:
The Kubernetes API server provides 3 API endpoints (healthz, livez and readyz) to indicate the current status of the API server. The healthz endpoint is deprecated (since Kubernetes v1.16), and you should use the more specific livez and readyz endpoints instead.